projects/ngx-d3timeline/src/lib/axis/serie-axis-view-model.ts

```diff
@@ -0,0 +1,37 @@
+import { ScaleBand, scaleBand } from 'd3-scale';
+import { TimelineEvent } from '../timeline-event';
+import { View } from '../view/view';
+
+export class SeriesAxisViewModel {
+  private readonly bandscale: ScaleBand<string>;
+
+  constructor(data: TimelineEvent[], view: View) {
+    this.bandscale = this.createBandscale(data, view);
+  }
+
+  get tickInfo(): { tick: string; tickTransform: string }[] {
+    return this.bandscale.domain().map(tick => ({
+      tick,
+      tickTransform: this.tickTransform(tick)
+    }));
+  }
+
+  get rangeLimit() {
+    return this.bandscale.range()[1];
+  }
+
+  get bandwidth() {
+    return this.bandscale.bandwidth();
+  }
+
+  private createBandscale(data: TimelineEvent[], view: View) {
+    return scaleBand()
+      .domain([...new Set(data.map(d => d.series))])
+      .range([0, view.width - view.margin]);
```

Comment:
move left/right extreme calculation logic to new TimelineView class also with height, width, margin properties

```suggestion
      .range([timelineView.bounds.left, timelineView.bounds.right]);
```

Reply:
done